src/common/safe_io.c

```diff
@@ -43,6 +44,30 @@ ssize_t safe_read(int fd, void *buf, size_t count)
 	return cnt;
 }
 
+// "read" doesn't work with Windows sockets.
+ssize_t safe_recv(int fd, void *buf, size_t count)
```

Comment:
Nit: similar comment for these new functions. If the client code is using sockets, perhaps it should always call `safe_recv`/`safe_send` and this code properly handles the Windows vs POSIX case.